api4/team.go

```diff
@@ -1275,12 +1275,23 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	emailList := model.ArrayFromJSON(r.Body)
+	memberInvite := &model.MemberInvite{}
+	if jsonErr := json.NewDecoder(r.Body).Decode(memberInvite); jsonErr != nil {
+		c.Err = model.NewAppError("Api4.inviteUsersToTeams", "api.team.invite_members_to_team_and_channels.invalid_body.app_error", nil, jsonErr.Error(), http.StatusBadRequest)
+		return
+	}
+
+	emailList := memberInvite.Emails
 
 	for i := range emailList {
 		emailList[i] = strings.ToLower(emailList[i])
 	}
 
+	if err := memberInvite.IsValid(); err != nil {
+		c.Err = err
+		return
+	}
+
 	if len(emailList) == 0 {
 		c.SetInvalidParam("user_email")
 		return
```

Comment:
Additionally, it seems that this is included in the valid check above so we should converge in checking in one place.